llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -1168,16 +1165,13 @@ public void allocateTask(Object task, Resource capability, ContainerId container
         capability, null, null, clock.getTime(), id);
     LOG.info("Received allocateRequest. task={}, priority={}, capability={}, containerId={}",
         task, priority, capability, containerId);
-    writeLock.lock();
-    try {
-      if (!dagRunning && metrics != null && id != null) {
+    if (!dagRunning) {
+      if (metrics != null && id != null) {
```

Comment:
One issue i can see is , we are setting the dagRunning to true without setting any stats. So the stats reading thread may get some junk value for numRequestedAllocations etc. is it fine ?

Reply:
Thanks for checking this out Mahesh, I guess this behaviour was already existing -- what changed in this PR is avoiding the writeLock for dagRunning.
Since now metrics like **numRequestedAllocations** and **numRequestsWithLocation** are Atomic* the writelock can be removed.

Reply:
makes sense. +1 from my side.